### DISCUSSION TITLE
WIth a microapp setup, what is the suggested way of migrating from v5 <router history={history} /> to <RouterProvider router={router} /> ?

### DISCUSSION THREAD

**Discussion by lundha** [remix-run/react-router#11678] [v5 to v6 Migration]
Hi,

In my current setup I have a base app running react-router-dom v5 and multiple microapps running the same version. 
I currently have the <Microapp /> components inside a <BrowserRouter /> in the host app. However these apps are hosted separately and mounted in. I inject the history object from useHistory() into the <Router /> in these apps. 

AFAIK, when migrating to V6, this is not possible unless I use the unstable_Router, in both the host app and the remote apps, and also injects the object from createBrowserHistory(). 

If not, when using a separate <BrowserRouter /> in the micro apps, the sub routes are not rendered, since it seems that the changes to the history are not detected. 

What are the recommended way going forward here? I want to use the <RouterProvider /> and the data load API. Or at least do routing without using the unstable_router from.

**Comment by slivingston-isn** [comment]
We have the same problem in v7. Did you ever figure out a way to do this?
